src/Models/AZ3166Device.ts

```diff
@@ -168,7 +172,21 @@ export class AZ3166Device extends ArduinoDeviceBase {
       return retValue;
     } else if (configSelection.detail === "Config Connection String") {
       // Get IoT Hub device connection string from config
-      let deviceConnectionString = ConfigHandler.get<string>(ConfigKey.iotHubDeviceConnectionString);
+      const componentConfigs = await this.azureConfigFileHandler.getSortedComponents(ScaffoldType.Workspace);

```

Comment:
What is the purpose of getSortedComponents method? Since I didn't seen any code change about AzureConfigFileHandler, I assume you just use the current implementation of that class. You could check if the current AzureConfigFileHandler methods are well designed and tailor it in a proper way.

Reply:
Already changed to use another new-added method to get componentConfig.